code/datums/elements/dryable.dm

```diff
@@ -20,32 +20,37 @@
 	UnregisterSignal(target, COMSIG_FOOD_CONSUMED)
 	REMOVE_TRAIT(target, TRAIT_DRYABLE, ELEMENT_TRAIT(type))
 
-/datum/element/dryable/proc/finish_drying(atom/source)
+/datum/element/dryable/proc/finish_drying(atom/source, datum/weakref/drying_user)
 	SIGNAL_HANDLER
 	var/atom/dried_atom = source
 	if(dry_result == dried_atom.type)//if the dried type is the same as our currrent state, don't bother creating a whole new item, just re-color it.
 		var/atom/movable/resulting_atom = dried_atom
 		resulting_atom.add_atom_colour(COLOR_DRIED_TAN, FIXED_COLOUR_PRIORITY)
-		ADD_TRAIT(resulting_atom, TRAIT_DRIED, ELEMENT_TRAIT(type))
+		apply_dried_status(resulting_atom, drying_user)
 		resulting_atom.forceMove(source.drop_location())
 		return
 	else if(isstack(source)) //Check if its a sheet
 		var/obj/item/stack/itemstack = dried_atom
 		for(var/i in 1 to itemstack.amount)
 			var/atom/movable/resulting_atom = new dry_result(source.drop_location())
-			ADD_TRAIT(resulting_atom, TRAIT_DRIED, ELEMENT_TRAIT(type))
+			apply_dried_status(resulting_atom, drying_user)
 		qdel(source)
 		return
 	else if(istype(source, /obj/item/food) && ispath(dry_result, /obj/item/food))
 		var/obj/item/food/source_food = source
 		var/obj/item/food/resulting_food = new dry_result(source.drop_location())
 		resulting_food.reagents.clear_reagents()
 		source_food.reagents.trans_to(resulting_food, source_food.reagents.total_volume)
-		ADD_TRAIT(resulting_food, TRAIT_DRIED, ELEMENT_TRAIT(type))
+		apply_dried_status(resulting_food, drying_user)
 		qdel(source)
 		return
 	else
 		var/atom/movable/resulting_atom = new dry_result(source.drop_location())
-		ADD_TRAIT(resulting_atom, TRAIT_DRIED, ELEMENT_TRAIT(type))
+		apply_dried_status(resulting_atom, drying_user)
 		qdel(source)
 
+/datum/element/dryable/proc/apply_dried_status(atom/target, datum/weakref/drying_user)
+	ADD_TRAIT(target, TRAIT_DRIED, ELEMENT_TRAIT(type))
+	var/datum/mind/user_mind = drying_user?.resolve()
+	if(drying_user && istype(target, /obj/item/food))
```

Comment:
I don't like type checking for food item, food is a component so we should be using that instead, no?

Reply:
Eedible is the component, while food is the item. Beside, the chef made trait is only useful for food items, which have the recipe complexity variable that dictates the duration and strength of the effects of eaten food (so far only an imperceptible speed boost, and shock immunity for high tier stuff).

TL;DR, the trait is only relevant for food items so far.

Reply:
However, I believe an `isfood()` macro should be made. It's beyond the scope of this PR.

Reply:
That's dumb it should've been on the component why are they killing me

Reply:
It's The Foodening, bro.